src/screens/Home/HomeScreen.js

```diff
@@ -23,11 +27,54 @@ const styles = StyleSheet.create({
 export default function HomeScreen({ navigation }) {
   const [initializing, setInitializing] = useState(true);
   const [user, setUser] = useState();
+  const [userState, setUserState] = useState(INITIAL_USER_STATE);
   const [uid, setUid] = useState(navigation.getParam('uid', null));
+  const ref = Firestore().collection('users');
+  const [code, setCode] = useState('');
+
+  async function getUserData() {
+    try {
+      const doc = await ref.doc(uid).get();
+      const data = doc.data();
+      // Handler for case with nonexistent user entries in Firestore
+      setUserState(data);
+      if (initializing) setInitializing(false);
+    } catch (e) {
+      setInitializing(false);
+      Alert.alert(
+        e.message,
+      );
+    }
+  }
+
+  useEffect(() => {
+    getUserData();
+  }, [userState]);
 
   function onAuthStateChanged(authUser) {
     setUser(authUser);
-    if (initializing) setInitializing(false);
+  }
```

Comment:
I think the code that you added to HomeScreen.js would be better suited somewhere else (I will have to think about it and talk to designers). Creating and joining a classroom would definitely need to happen during sign up process, so you could work on that for your next sprint.

Reply:
Sounds good - I left it there at the moment because I figured it would be useful to build and easy to test on the Home Screen.